File: 33.투 포인터/1806.부분합/main.py
import sys

input = sys.stdin.readline

if __name__ == "__main__":
    N, S = map(int, input().split())
    lst = list(map(int, input().split()))
    left, right = 0, 0

    cur_sum = lst[0]
    min_len = N + 1
    while left <= right:
        if cur_sum >= S:
            min_len = min(min_len, right - left + 1)
            cur_sum -= lst[left]
            left += 1
        else:
            right += 1
            if right == N:
                break
            cur_sum += lst[right]

    print(min_len if min_len != N + 1 else 0)

File: 33.투 포인터/1806.부분합/Main.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int S = Integer.parseInt(st.nextToken());

        int[] arr = new int[N];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < arr.length; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        int left = 0, right = 0;
        int curSum = arr[0];
        int result = N + 1;
        while (left <= right) {
            if (curSum >= S) {
                result = Math.min(right - left + 1, result);
                curSum -= arr[left++];
            } else {
                right += 1;
                if (right == N)
                    break;
                curSum += arr[right];
            }
        }

        System.out.println(result != N + 1 ? result : "0");
    }
}